### ISSUE TITLE
Unable to install Intel SGX DCAP driver on Ubuntu 20.04 with kernel 5.15.0-1035-azure

### ISSUE THREAD

**Issue by troublesprouter** [intel/linux-sgx#945]
I am trying to install the Intel SGX DCAP driver on a virtual machine running Ubuntu 20.04 with kernel version 5.15.0-1035-azure. However, the installation fails with the following error message:

```
Failed to build driver.
DKMS make.log for sgx-1.41 for kernel 5.15.0-1035-azure (x86_64)
Wed Apr 12 22:46:31 UTC 2023
Makefile:24: *** Can't install DCAP SGX driver with inkernel SGX support.  Stop.
I have followed the suggested steps to disable the in-kernel SGX driver, such as blacklisting intel_sgx in /etc/modprobe.d/blacklist.conf and adding noxsave and nosgx to the kernel parameters in /etc/default/grub. However, the issue persists.
```

Here is the output of `uname -a`:

`Linux accvm3 5.15.0-1035-azure #42~20.04.1-Ubuntu SMP Wed Mar 1 19:17:41 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
`

And the output of `sudo dmesg | grep -i sgx`:
```
[    0.550821] sgx: EPC section 0x1c0200000-0x1c1bfffff
[    0.552345] sgx: EPC section 0x1c1e00000-0x1c1ffffff
```

I would appreciate any help or suggestions to resolve this issue.

The reason why I need to do this is that I need ECDSA signatures and the readme says:

> To use ECDSA attestation, you must install Intel(R) Software Guard Extensions Driver for Data Center Attestation Primitives (Intel(R) SGX DCAP). Please follow the [Intel(R) SGX DCAP Installation Guide for Linux* OS](https://download.01.org/intel-sgx/latest/dcap-latest/linux/docs/Intel_SGX_SW_Installation_Guide_for_Linux.pdf) to install the Intel(R) SGX DCAP driver.
> NOTE: If you had already installed Intel(R) SGX driver without ECDSA attestation, please uninstall the driver firstly and then install the Intel(R) SGX DCAP driver. Otherwise the newly installed Intel(R) SGX DCAP driver will be unworkable.

**Comment by llly** [comment]
You don't need to installer any SGX driver nor disable in-kernel SGX driver for 5.15.0-1035-azure to use ECDSA attestation.

**Comment by troublesprouter** [comment]
Thanks. I suggest updating the readme to reflect that.

Using this Azure Confidential VM setup, can I just consider these three [prerequisites](https://github.com/apache/incubator-teaclave-sgx-sdk#prerequisites) from Teaclave met by default?

> Intel SGX OOT 2.11.0 Driver or DCAP 1.36.2 Driver
> Intel SGX SDK v2.12
> Intel SGX PSW

Do I instead only need to run t[hese two commands](https://github.com/intel/linux-sgx#quick-start-with-docker-and-docker-compose) and then go straight to installing Teaclave from there?:

> Quick Start with Docker and Docker Compose
> Build PSW and SDK from source. See this [README](https://github.com/intel/linux-sgx/blob/master/docker/build/README.md) for details.
> `$ cd docker/build && ./build_compose_run.sh`

Just seeking the simplest, shortest path to running my Rust script in the enclave and get it ECDSA attested in the enclave.

Appreciated.

**Comment by ScottR-Intel** [comment]
@troublesprouter  ACC VMs only provide the SGX driver.  Everything else required (SDK, PSW) has to be installed by the user.  You don't need to build the SDK and PSW yourself...  you could just install from our repo as described in the Linux Install Guide you linked to earlier (skipping the driver install parts, of course).  After that, you should be able to start the Teaclave install.

**Comment by ScottR-Intel** [comment]
Apologies... Yes, you absolutely need the DCAP components also... I should have said "SDK, PSW, DCAP".  

No, since you're running in ACC, Ubuntu 20.04 is fine because it has the in-kernel driver built in because Microsoft uses a special 20.04 build.  If you were running on your own system, I'd recommend Ubuntu 22.04 because it's the first version that had the kernel with the SGX in-kernel driver out of the box, though you could also install the Ubuntu HWE package for Ubuntu 20.04 also and get the SGX in-kernel driver.

**Comment by ScottR-Intel** [comment]
Correct... you do not need to install your own PCCS... You will ultimately use the Microsoft THIM service, which is their version of a PCCS.  For Ubuntu 20.04, you can use Microsoft's [Azure DCAP Client](https://github.com/microsoft/Azure-DCAP-Client/releases) instead of Intel's QPL/QCNL packages.  More info [here](https://learn.microsoft.com/en-us/azure/confidential-computing/quick-create-portal) and [here](https://learn.microsoft.com/en-us/azure/security/fundamentals/trusted-hardware-identity-management#what-is-the-azure-dcap-library).
